core/domain/exp_domain.py

```diff
@@ -2444,12 +2501,11 @@ def get_state_names_mapping(self, change_list):
         for state_name in self.states:
             old_to_new_state_names[state_name] = state_name
 
-        for change_dict in reversed(change_list):
-            if change_dict['cmd'] == CMD_RENAME_STATE:
-                if change_dict['new_state_name'] in old_to_new_state_names:
-                    old_to_new_state_names[change_dict['old_state_name']] = (
-                        old_to_new_state_names.pop(change_dict[
-                            'new_state_name']))
+        exp_versions_diff = ExplorationVersionsDiff(change_list)
```

Comment:
Hi @pranavsid98 -- I thought we chatted about re-examining this method because we noticed it didn't take into account adds and deletes? Could you please check with @prasanna08 about that and verify that its usage is correct? (With your new method it might be possible to simplify this, too, and perhaps even get rid of get_state_names_mapping() altogether.)

Reply:
Done.